ui.tests/test-module/specs/wizard/wizard.runtime.spec.js

```diff
@@ -309,22 +277,60 @@ describe("Form with wizard Layout Container with Hidden Children", () => {
+                        const thirdItemOfWizardId = formContainer._model.items[0].items[2].id + "__wizardpanel";
+                        cy.get(`#${thirdItemNavOfWizardId}`).should('have.class', 'cmp-adaptiveform-wizard__tab--active');
+                        cy.get(`#${thirdItemOfWizardId}`).should('have.class', 'cmp-adaptiveform-wizard__wizardpanel--active');
                         cy.get('.cmp-adaptiveform-wizard__nextNav').click({force: true}).then(() => {
-                            cy.get(`#${thirdItemNavOfWizardId}`).should('have.class', 'cmp-adaptiveform-wizard__tab--active');
-                            cy.get(`#${thirdItemOfWizardId}`).should('have.class', 'cmp-adaptiveform-wizard__wizardpanel--active');
+                            cy.get(`#${fourthItemNavOfWizardId}`).should('have.class', 'cmp-adaptiveform-wizard__tab--active');
+                            cy.get(`#${fourthItemOfWizardId}`).should('have.class', 'cmp-adaptiveform-wizard__wizardpanel--active');
                         })
                     })
                 })
             })
         })
     });
+
+    it("rule based visibility of tabs is toggles visibility of nav buttons", () => {
+        const textInputId = formContainer._model.items[1].id;
+        const wizardItems = formContainer._model.items[0].items;
+
+        const firstItemNavOfWizardId = wizardItems[0].id + "_wizard-item-nav",
+            firstItemOfWizardId = wizardItems[0].id + "__wizardpanel",
+            secondItemNavOfWizardId = wizardItems[1].id + "_wizard-item-nav",
+            secondItemOfWizardId = wizardItems[1].id + "__wizardpanel";
+
+        const previousNavButton = '.cmp-adaptiveform-wizard__previousNav',
+            nextNavButton = '.cmp-adaptiveform-wizard__nextNav';
+
+        cy.get(`#${firstItemNavOfWizardId}`).should('have.attr', 'data-cmp-visible', 'false');
+        cy.get(`#${firstItemOfWizardId}`).should('have.attr', 'aria-hidden', 'true');
+        cy.get(`#${secondItemNavOfWizardId}`).should('have.class', 'cmp-adaptiveform-wizard__tab--active');
+        cy.get(`#${secondItemOfWizardId}`).should('have.class', 'cmp-adaptiveform-wizard__wizardpanel--active');
+        cy.get(previousNavButton).should('not.be.visible').should('have.attr', 'data-cmp-visible', 'false');
+        cy.get(nextNavButton).should('exist');
+
+        cy.get(`#${textInputId}`).find('.cmp-adaptiveform-textinput__widget').focus().type('b').blur().then(() => {
+            cy.get(`#${secondItemNavOfWizardId}`).should('have.attr', 'data-cmp-visible', 'false');
+            cy.get(`#${secondItemOfWizardId}`).should('have.attr', 'aria-hidden', 'true');
+            cy.get(previousNavButton).should('not.be.visible').should('have.attr', 'data-cmp-visible', 'false');
+
+            cy.get(`#${textInputId}`).find('.cmp-adaptiveform-textinput__widget').focus().clear().type('a').blur().then(() => {
+                cy.get(previousNavButton).should('exist');
```

Comment:
why are we checking should 'exist', 'exist' checks present in dom.
We should be checking be.visible.